token-lending/program/tests/helpers/mod.rs

```diff
@@ -587,7 +591,12 @@ impl TestLendingMarket {
             ],
             recent_blockhash,
         );
-        assert!(banks_client.process_transaction(transaction).await.is_ok());
+        let result = banks_client.process_transaction(transaction).await;
+        if successful_transaction {
+            assert!(result.is_ok());
+        }
```

Comment:
I would just pass the result through. Whenever a function in Rust returns a `Result`, the compiler will bug you to check the result anyways so we can chat for `ok` there